### ISSUE TITLE
allow to create an optional event on max bound

### ISSUE THREAD

**Issue by theod** [ossia/score#142]
I keep this proposition as a question as it need to be discussed together.
it allows to create alternative branch depending if the interaction happen or not before the max bound.
the idea have been fully described on the dev list.

**Comment by theod** [comment]
this depends on a feature of API : https://github.com/OSSIA/API/issues/22

**Comment by nvuaille** [comment]
I realize yesterday that one can use 'fake' osc device as an internal memory to store global var. So I was able to create a scenario resolving the blue yeti problem (a player run and press a button at the end. A short delay means cheat, a too long one means player left) with a system of flag.

That's cool but not very user friendly, I agree.

But I think we could improve that by using intern var / signals reporting trigger status, and catch it in condition expression. So in the end node you just have to set events condition with 'if triggered' or 'if not triggered'.
This solution will not add a new element in the syntax of i-score and will allow the same things as the proposal element.

**Comment by theod** [comment]
yes the solution you proposed was quite the same than Blue Yeti proposed in the Runner example (that you can read here : https://www.dropbox.com/s/8prlpioqyg750y2/M_%20use%20case%20Arena.pdf?dl=0 ). in this solution Blue Yeti resolved this case by comparing the elapsed time of the constraint and its max duration bound (considering i-score elements as part of a local device).

I understand your reluctance to add a new element to the formalism and this always have to be done cautiously. however the solution we discussed with @bltzr, @jcelerier and @blueyeti to write the JIM paper have the great advantage to propose a very readable solution that avoid users to look into a local device to understand what is going on.

IHMO that is typically the kind of problem that i-score have to solve ingeniously if we want to provide a tool that makes people's life easier when they deal with those kinds of logics-temporal authoring issue.

I'll forward you the article in a private email.

**Comment by theod** [comment]
@bltzr found a temporary solution using a local boolean variable set to true when the trigger is triggered. then on the timenode there is 2 conditions : if variable == false (which is the case when the trigger has not been triggered) or if variable == true (the trigger has been triggered).

IMHO it is not final solution
